vcd/resource_vcd_catalog_media.go

```diff
@@ -152,19 +191,40 @@ func resourceVcdMediaRead(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf(errorRetrievingOrg, err)
 	}
 
-	mediaItem, err := vdc.FindMediaImage(d.Get("name").(string))
-	if err != nil || mediaItem == (govcd.MediaItem{}) {
+	catalogItem, err := findCatalogItem(d, meta.(*VCDClient))
+	if err != nil {
+		log.Printf("[DEBUG] Unable to find media item: %s", err)
+		return err
+	}
+	if catalogItem == nil {
+		log.Printf("[DEBUG] Unable to find media item: %s. Removing from tfstate", err)
+		return nil
+	}
+
+	mediaItem, err := vdc.QueryMediaImage(catalogItem.CatalogItem.Name, d.Get("catalog").(string))
```

Comment:
This call will fail. 
There is no VDC in the media arguments, and there is no guarantee that there will be such parameter in the provider section, since we can use catalogs and media without mentioning a VDC.
Please use `Catalog.GetMedia*` to retrieve the image

Reply:
it's not an option cause Catalog.GetMedia don't return all attributes I need

Reply:
changed